BigBaseV2/src/gta/joaat.hpp

```diff
@@ -7,78 +7,32 @@
 namespace rage
 {
 	using joaat_t = std::uint32_t;
-
-	inline constexpr char joaat_to_lower(char c)
-	{
-		return (c >= 'A' && c <= 'Z') ? c + ('a' - 'A') : c;
-	}
+	inline constexpr char joaat_to_lower(char c) { return (c >= 'A' && c <= 'Z') ? c + ('a' - 'A') : c; }
 
 	template <std::size_t CharCount>
-	struct constexpr_joaat
-	{
+	struct constexpr_joaat {
 		char data[CharCount];
-
 		template <std::size_t... Indices>
-		constexpr constexpr_joaat(const char *str, std::index_sequence<Indices...>) :
-			data{ (str[Indices])... }
-		{
-		}
-
+		constexpr constexpr_joaat(const char *str, std::index_sequence<Indices...>) : data{ (str[Indices])... } {}
 		constexpr joaat_t operator()()
 		{
 			joaat_t hash = 0;
-
 			for (std::size_t i = 0; i < CharCount; ++i)
-			{
-				hash += joaat_to_lower(data[i]);
-				hash += (hash << 10);
-				hash ^= (hash >> 6);
-			}
-
-			hash += (hash << 3);
-			hash ^= (hash >> 11);
-			hash += (hash << 15);
-
+				hash += joaat_to_lower(data[i]), hash += (hash << 10), hash ^= (hash >> 6);
+			hash += (hash << 3), hash ^= (hash >> 11), hash += (hash << 15);
 			return hash;
 		}
 	};
 
 	inline joaat_t joaat(std::string_view str)
 	{
 		joaat_t hash = 0;
-
-		for (char c : str)
-		{
-			hash += joaat_to_lower(c);
-			hash += (hash << 10);
-			hash ^= (hash >> 6);
-		}
-
-		hash += (hash << 3);
-		hash ^= (hash >> 11);
-		hash += (hash << 15);
-
-		return hash;
-	}
-
-	inline joaat_t joaat(const char *str)
-	{
-		joaat_t hash = 0;
-
-		while (*str)
-		{
-			hash += joaat_to_lower(*(str++));
-			hash += (hash << 10);
-			hash ^= (hash >> 6);
-		}
-
-		hash += (hash << 3);
-		hash ^= (hash >> 11);
-		hash += (hash << 15);
-
+		for (auto c : str)
+			hash += joaat_to_lower(c), hash += (hash << 10), hash ^= (hash >> 6);
+		hash += (hash << 3), hash ^= (hash >> 11), hash += (hash << 15);
```

Comment:
Reason why you rammed this all into a single line, it's good that you removed the redundant `inline joaat_t joaat(const char *str)` but sacrificing readability for something that compiles to the same assembly seems unnecessary.

Reply:
Fixed